connection.go

```diff
@@ -778,6 +778,12 @@ func (s *connection) handleHandshakeConfirmed() error {
 		}
 		s.mtuDiscoverer.Start(utils.Min(maxPacketSize, protocol.MaxPacketBufferSize))
 	}
+
+	// do not log earlier to not apply additional pressure on the handshake mutex
+	if s.tracer != nil && s.tracer.ChoseAlpn != nil {
```

Comment:
We should be able to log this earlier, at the very latest on handshake completion.